src/IP6BeamPipe.cpp

```diff
@@ -34,35 +35,48 @@ using namespace dd4hep;
  *
  */
 static Ref_t create_detector(Detector& det, xml_h e, SensitiveDetector /* sens */) {
-
   using namespace ROOT::Math;
   xml_det_t x_det       = e;
   string det_name       = x_det.nameStr();
   xml_comp_t x_dettype  = x_det.child(dd4hep::xml::Strng_t("type_flags"));
   unsigned int typeFlag = x_dettype.type();
   DetElement sdet(det_name, x_det.id());
   Assembly assembly(det_name + "_assembly");
-  Material m_Al     = det.material("Aluminum");
-  Material m_Be     = det.material("Beryllium");
-  Material m_Au     = det.material("Gold");
-  Material m_Vacuum = det.material("Vacuum");
-  string vis_name   = x_det.visStr();
 
   xml::Component IP_pipe_c = x_det.child(_Unicode(IP_pipe));
 
-  // IP
-  double IP_beampipe_OD             = IP_pipe_c.attr<double>(_Unicode(OD));
-  double IP_beampipe_wall_thickness = IP_pipe_c.attr<double>(_Unicode(wall_thickness));
-  double IP_beampipe_gold_thickness = IP_pipe_c.attr<double>(_Unicode(gold_thickness));
-  double IP_beampipe_ID =
-      IP_beampipe_OD - 2.0 * IP_beampipe_gold_thickness - 2.0 * IP_beampipe_wall_thickness;
-  double IP_acts_beampipe_OD = IP_beampipe_ID - 5.0 * mm;
-  double IP_acts_beampipe_ID = IP_acts_beampipe_OD - 1.0 * mm;
+  Material IP_beampipe_wall_material =
+      det.material(IP_pipe_c.attr<string>(_Unicode(wall_material)));
+  Material IP_beampipe_coating_material =
+      det.material(IP_pipe_c.attr<string>(_Unicode(coating_material)));
+  Material m_Vacuum  = det.material("Vacuum");
+  Material m_Wall    = det.material("StainlessSteel");
+  Material m_Coating = det.material("Copper");
+  string vis_name    = x_det.visStr();
+
+  // IP Beampipe
+  double IP_beampipe_ID                = IP_pipe_c.attr<double>(_Unicode(ID));
+  double IP_beampipe_wall_thickness    = IP_pipe_c.attr<double>(_Unicode(wall_thickness));
+  double IP_beampipe_coating_thickness = IP_pipe_c.attr<double>(_Unicode(coating_thickness));
+  double IP_acts_beampipe_OD           = IP_beampipe_ID - 5.0 * mm;
+  double IP_acts_beampipe_ID           = IP_acts_beampipe_OD - 1.0 * mm;
```

Comment:
Check with @ShujieL  on these ACTS beampipe settings. Has this been validated in reconstruction? Is there a plan for someone to create a new ACTS material map?

Reply:
What are 5 and 1 mm here? We need a comment documenting these.